internal/receiver/smartagentreceiver/output_test.go

```diff
@@ -354,32 +354,29 @@ type hostWithExporters struct {
 	exporter *mockMetadataClient
 }
 
-func getExporters() map[config.DataType]map[config.NamedEntity]component.Exporter {
-	exporters := map[config.DataType]map[config.NamedEntity]component.Exporter{}
-	metricExporterMap := map[config.NamedEntity]component.Exporter{}
+func getExporters() map[config.DataType]map[config.ComponentID]component.Exporter {
+	exporters := map[config.DataType]map[config.ComponentID]component.Exporter{}
+	metricExporterMap := map[config.ComponentID]component.Exporter{}
 	exporters[config.MetricsDataType] = metricExporterMap
 
 	exampleExporterFactory := componenttest.NewNopExporterFactory()
 	exampleExporter, _ := exampleExporterFactory.CreateMetricsExporter(
 		context.Background(), component.ExporterCreateParams{}, nil,
 	)
-	metricExporterMap[exampleExporterFactory.CreateDefaultConfig()] = exampleExporter
 
-	receiver := namedEntity{name: "metricsreceiver"}
-	metricExporterMap[&receiver] = &mockMetricsReceiver{}
-
-	notReceiver := namedEntity{name: "notareceiver"}
-	metricExporterMap[&notReceiver] = &notAReceiver{}
+	metricExporterMap[exampleExporterFactory.CreateDefaultConfig().ID()] = exampleExporter
+	metricExporterMap[config.NewID("metricsreceiver")] = &mockMetricsReceiver{}
```

Comment:
I believe these `ComponentID`'s need names in order to provide something to match against: https://github.com/signalfx/splunk-otel-collector/blob/237af7d347ded7f42e834cbb2018f694346cf5dd/internal/receiver/smartagentreceiver/output.go#L132

Reply:
Done. In the new `ComponentID` the `.String()` is the equivalent with the previous `Name()` and `Name()` is just the part without type.

I fixed this by doing the right thing in the output.go. Thanks for catching this subtle bug.